### ISSUE TITLE
Nested query on has_one with descendants not returning union type

### ISSUE THREAD

**Issue by emmettdn** [silverstripe/silverstripe-graphql#136]
When querying a has_one relationship in a nested query, the type returned is the base type instead of a union type.

If an object "Item" has_one "Service" and "Specific_Service" extends "Service", the following query 

```
query {
  readItems {
    Service {
        __typename
    }
  }
}
```

Returns

```
{
  "data": {
    "readItems": [
      {
        "Service": {
          "__typename": "Service"
        }
      }
    ]
  }
}
```

Expected

```
{
  "data": {
    "readItems": [
      {
        "Service": {
          "__typename": "ServiceWithDescendants"
        }
      }
    ]
  }
}
```

## Pull Requests
- [ ] https://github.com/silverstripe/silverstripe-graphql/pull/150

**Comment by flamerohr** [comment]
from @unclecheese on Slack
> has_ones do not use union types
> probably also true for nested queries
> (has_many /many_many)

**Comment by emmettdn** [comment]
Is this something that might be looked at in the near future?
If not, I would be happy to attempt to help out with a PR but would anyone be able to point me in the right direction?

**Comment by tractorcow** [comment]
Initial review left.

**Comment by tractorcow** [comment]
Fixed with #150 thanks @unclecheese !

**Comment by altwohill** [comment]
How do we get a nested query (has_many) to scaffold a union type? I can't tell from the docs
